publication-commons/src/main/java/no/unit/nva/publication/model/business/PublishingRequestCase.java

```diff
@@ -283,6 +290,72 @@ public PublishingRequestCase approveFiles() {
         return this;
     }
 
+    public void publishApprovedFiles(ResourceService resourceService) {
+        if (resourceService.shouldUseNewFiles()) {
+            getApprovedFiles().forEach(file ->
+                                           FileEntry.queryObject(file.getIdentifier(), getResourceIdentifier())
+                                               .fetch(resourceService)
+                                               .ifPresent(fileEntry -> fileEntry.approve(resourceService)));
+        } else {
+            var resource = Resource.resourceQueryObject(getResourceIdentifier()).fetch(resourceService).orElseThrow();
+            var resourceWithUpdatedAssociatedArtifacts = toPublicationWithApprovedFiles(resource).toPublication();
+            resourceService.updatePublication(resourceWithUpdatedAssociatedArtifacts);
+        }
+    }
+
+    public void rejectRejectedFiles(ResourceService resourceService) {
+        if (resourceService.shouldUseNewFiles()) {
+            getFilesForApproval().stream()
```

Comment:
maybe we should rename getFilesForApproval to something that does not assume that is should be always approved. getFilesForEvaluation()